### ISSUE TITLE
Refactor `:detekt-tooling`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#7014]
## Expected Behavior
`:detekt-tooling` is the public api for anyone that wants to implement a `detekt` client

## Current Behavior
Right now we have a mixture between `:detekt-tooling` and `:detekt-cli`. And also `:detekt-tooling` has things that shouldn't be part of the API.

## Context
I think that `:detekt-gradle-plugin` shouldn't use `:detekt-cli`. And it should depend on `:detekt-core`. But, because we don't want to make `:detekt-core` public, we can use `:detekt-tooling` as the public api containing only the `interface`s, `enum`s and similar classes.

**Comment by 3flex** [comment]
Agree, tooling API should be the entry point for all clients - CLI, Gradle plugin and compiler plugin. And it should be used by the IDE plugin if it's not already (but I think it was originally introduced for that purpose?)
